### PR TITLE
GetMultiRegionHost error is ignored leading to malformed multiregion resources

### PR DESCRIPTION
**Pull request by dikhan**
## Proposed changes

Please add as many details as possible about the change here. Does this Pull Request resolve any open issue? If so, please make sure to link to that issue:
                                                              
Fixes: #137 

## Type of change

What type of change does your code introduce to the provider? Please put an `x` (w/o heading/trailing white spaces) 
in the boxes that apply:

- [x] Bug-fix (change that fixes current functionality)
- [x] Tech debt
- [ ] New feature (change that adds new functionality)

## Checklist

Please put an `x` (w/o heading/trailing white spaces) in the boxes that apply:

- [x] I have read and followed the [CONTRIBUTING](https://github.com/dikhan/terraform-provider-api/blob/master/.github/CONTRIBUTING.md) guidelines
- [x] I have made sure code compiles correctly
- [x] I have run 'make test' locally from the terraform_provider_api folder and no errors were found
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added/updated necessary documentation (if appropriate)

### CODE REVIEW

File: openapi/openapi_v2_spec_analyser.go [hunk 1]
```diff
@@ -60,20 +73,12 @@ func (specAnalyser *specV2Analyser) GetTerraformCompliantResources() ([]SpecReso
...
-				if err != nil {
-					log.Printf("multi region host for resource '%s' is not valid: ", err)
-					continue
-				}
-				log.Printf("[INFO] multi region resource name = %s, region = '%s', host = '%s'", r.getResourceName(), regionName, regionHost)
-				resources = append(resources, r)
+			multiRegionResources, err := specAnalyser.createMultiRegionResources(regions, resourceRootPath, *resourceRoot, pathItem, resourcePayloadSchemaDef)
+			if err != nil {
+				log.Printf("[WARN] ignoring multiregion resource '%s' due to an error: %s", resourceRootPath, err)
+				continue
```
Comment (jccisco) [inline openapi/openapi_v2_spec_analyser.go]: I would expect that if there's an error from the createMultiRegionResources calls, then multiRegionResources will be an empty slice, and then can safely go ahead with the following `append` call, meaning that we can drop the `continue`.  At that point, if it's not important enough to guard against a regression in the log call, perhaps we can just eliminate the entire `err != nil` conditional branch.

**Comment by dikhan** [comment]
@jccisco  I believe we have already met the acceptance criteria for this story which involved adding more test coverage for GetTerraformCompliantResources. As part of that effort there's been some refactoring done to make it easier to test the function getting the code to a better shape to how it was before.

I would be mindful in terms on how much extra time/effort we want to spend on this card considering that at this point we have met the goal of the original request. If there's something that stands out that is critical or a major concern I'd agree on continuing working on this PR; otherwise I would be more inclined to getting this PR merged and pick up the next thing that will deliver value to the customer.

**Comment by jccisco** [comment]
Just to add to my comment above: I'm not an absolutist about this. If there is an unusual business need to get a bug fixed or a feature delivered ASAP, then I would be ok with flagging tech debt and leaving it for later.  But if there's no unusual deadline, we need to be continually maintain and improving code quality.

**Comment by dikhan** [comment]
Thanks, my position is always to meet the acceptance criteria, and while on it live by refactoring improving the quality of code to be more maintainable; understandable; well-documented by its tests AND any other customer documentation that might be needed; and well-defended against regression.

On the other hand, I believe you can do all the above without increasing the scope of stories to a point where it's no longer clear when to stop because at that point the definition of done comes down to individual judgement. 

To put an example:

I don't think every refactoring needs another card, however I believe this story shows clearly that there been a fair amount of work done here including a bug fix and some fair bit of refactoring along the way and originally it was suggested to bake this whole work as part of the sub-resource support story which would have increased it's scope greatly. This story was suggested and created instead of doing that, and as a result we have a clear description of the problem we are trying to solve which in turn is better surfaced so the community is aware of the bug and the fix put in place and also the scope and criteria are well defined.

**Comment by jccisco** [comment]
I need to apologize for being pedantic.  I hadn't pulled the latest when I wrote the above.  I see that the refactoring I had in mind was already done-- pulling the inner multiregion loop out to its own function; and that the new testcases document all the branches.  It looks good, and it's a good lesson to me to double check the validity of my assumptions before commenting.

**Pull request merged**